In [1]:
%matplotlib inline

In [2]:
import matplotlib.pyplot as plt
import pandas as pd
import re

# English Vocabulary Compliments of [dwyl](https://github.com/dwyl)
https://github.com/dwyl/english-words.git

In [3]:
def read_words_alpha():
    lines = []
    with open('./data/words/words_alpha.txt', 'r') as f:
        for _, line in enumerate(f):
            lines.append(line.strip())
            
    return lines

# From a to zwitterionic

In [4]:
words = read_words_alpha()

# Print out: n_words, head(10), tail(10)
len(words), words[:10], words[-10:]

(370103,
 ['a',
  'aa',
  'aaa',
  'aah',
  'aahed',
  'aahing',
  'aahs',
  'aal',
  'aalii',
  'aaliis'],
 ['zwanziger',
  'zwieback',
  'zwiebacks',
  'zwieselite',
  'zwinglian',
  'zwinglianism',
  'zwinglianist',
  'zwitter',
  'zwitterion',
  'zwitterionic'])

# Percent of words in vocabulary containg ie or ei

In [5]:
n_words_of_interest = len([w for w in words if 'ie' in w or 'ei' in w])

# Print the number of ie|ei words, % of ie|ei words
n_words_of_interest, n_words_of_interest / len(words)

(17542, 0.047397616339235296)

# Summary

Of **370k english words** only **4.7%** (17,542) contain ie|ei